### ISSUE TITLE
How to handle species outside of range

### ISSUE THREAD

**Issue by djhocking** [jclbrooks/MD_Stream_Salamanders#8]
E. guttolineata only occurs in the NCR (in VA but not north) and in Shenandoah but not in the rest of the range. Not sure how to handle this. E. longicauda, D. ochrophaeus, and G. porphyriticus may also be outside the range of NRC and on the edge of Shenandoah. D. Monticola not in range of NCR.

**Comment by jclbrooks** [comment]
We should do separate models for each species

*Bill Peterman suggested this because of the complexity that this issue can create within the model

See Email from Evan Grant from 8-9-19 that has both Evan's and Bill's responses on it for further details if needed

**Comment by djhocking** [comment]
multiple the whole psi formula by vector of species range

**Comment by djhocking** [comment]
What is the csv file or dataframe that has range (0/1) by transect?

**Comment by jclbrooks** [comment]
The object is spec_dist in line 57 of the format_landscape_data.R file

**Comment by djhocking** [comment]
stream mislabeled as transect so can't be used with other data

**Comment by djhocking** [comment]
And capitalization a problem

**Comment by djhocking** [comment]
Fixed I think and added to covariates in the `prep_occ.R` script.

```
featureids_matched <- read_csv("featureids_for_DAYMET_data.csv")
spp_dist <- read_csv("Data/species_ranges.csv") %>%
  rename(stream = transect) # transect mislabled as stream for western maryland

# fix the naming inconsistencies
transect_stream <- read_csv("Data/Location_Transect_Stream_Features_processed.csv") %>%
  select(stream = Stream, Transect) %>%
  distinct() %>%
  mutate(stream = ifelse(stream == "POPLICKTRIB", "PopLick", stream),
         stream = ifelse(stream == "ALEX", "Alexander Run", stream),
         stream = ifelse(stream == "ELKLICK", "ElkLick", stream),
         stream = ifelse(stream == "MILL", "Mill", stream),
         stream = ifelse(stream == "BLUELICK", "BlueLick", stream),
         stream = ifelse(stream == "WSHALEN", "West Shale North", stream),
         stream = ifelse(stream == "KOCH", "Koch", stream),
         stream = ifelse(stream == "DUNGHILL", "Bowser-Dung Hill", stream),
         stream = ifelse(stream == "BEARHILL", "Maynardier Ridge at Bear Hill", stream)) %>%
  mutate(transect = paste0(stream, "_", Transect))

# fix the stream vs. transect naming issues to be consistent and work with other data
spp_dist <- featureids_matched %>%
  select(-date) %>%
  distinct() %>%
  left_join(transect_stream) %>%
  mutate(stream = ifelse(region == "WMaryland", stream, transect)) %>%
  left_join(spp_dist) 

filter(spp_dist, WMaryland == 1) # check that worked as expected

spp_dist <- spp_dist %>%
  select(-Transect, -stream)

# write out object for later use
saveRDS(spp_dist, "Data/Derived/spp_dist.rds")
```

**Comment by djhocking** [comment]
filter out sites with NA because of bad lat lon

line 80 of `prep_occ.R`

**Comment by jclbrooks** [comment]
I filtered out the sites with the NAs in prep_occ.R . The transects that were removed were "COST28-2", "Fairway18-1", "Picnic 21-1", "COST8-3", and "W Shale North_1D". The first 4 were sites with bad lat/lons, and the last site had only one observation and visit. I'll make a master list with notes of the ones that were removed at some point in time for the Mid-Atlantic dynamic occupancy model.

commit cfe3f2d35912992abb711ae2a9b5e4e1d741f2b4

**Comment by djhocking** [comment]
Added $\zeta$ which is the "data" of whether the transect is within the species range or not (0/1).

```
 Z[i, t] ~ dbern(psi[i, t] * zeta[i]) 
```
This line in JAGS makes $\psi$ the suitability and $\psi * \zeta$ the prob of occupancy (within the species range).

All seems to be working.

**Comment by djhocking** [comment]
I think there was a problem with a couple species being caught outside their range so multiplying by $\zeta$ caused an incompatibility issue. We should check for either larger ranges or mis-identification (for publication not for thesis). Currently results are without range restriction for those (two?) species.

**Comment by jclbrooks** [comment]
It seems that the object 'df_occ_all' (created in line 95 of prep_occ.R) has not been expanded to all combinations of transect-year-species. There are 218 transects, 18 years, and 8 species; so there should theoretically be 31392 rows. Instead there are 4244 rows, a lot less. This seems to be because it didn't expand into species that were not recorded in the region (one example being: DMON, DOCH, and GPOR were not found in the National Capital Region and thus are not in this dataset and were never expanded to have NAs for these species). This is also a problem because we created the 3D arrays for each species that go into the model based off of 'df_occ_all'. Then again, maybe it sort of fixes itself because it's sort of a built-in version of Zeta??? I'm thinking this is why there were errors when you tried to multiply Zeta by each species array - because there weren't rows for each transect-species combination.

First off, do you think this is why the errors were throwing?? If it is, should I/we go ahead and expand the 'df_occ_all' to have all combinations so we can multiply by Zeta? Or just leave it as is without Zeta?

**Comment by djhocking** [comment]
Yeah, sorry I was hoping to try it out but haven't had a chance. This seems like a likely solution to the problem. Try it out and let me know if it works or if you get stuck. Thanks for tracking this down.
